src/Spice86.Core/Emulator/OperatingSystem/DosPathResolver.cs

```diff
@@ -0,0 +1,315 @@
+            }
+        }
+        return _driveMap[_currentDrive].DosDriveRootPath;
+    }
+
+    private string GetFullDosPathIncludingRoot(string absoluteOrRelativeDosPath) {
+        if(string.IsNullOrWhiteSpace(absoluteOrRelativeDosPath)) {
+            return absoluteOrRelativeDosPath;
+        }
+        StringBuilder normalizedDosPath = new();
+
+        string backslashedDosPath = ConvertUtils.ToBackSlashPath(absoluteOrRelativeDosPath);
+
+        string driveRoot = $"{GetDosDrivePathFromDosPath(backslashedDosPath)}{DirectorySeparatorChar}";
+        normalizedDosPath.Append(driveRoot);
+
+        if(backslashedDosPath.StartsWith(driveRoot)) {
+            backslashedDosPath = backslashedDosPath[3..];
+        }
+        else if (backslashedDosPath.StartsWith(driveRoot[..1])) {
+            backslashedDosPath = backslashedDosPath[2..];
+        }
+
+        IEnumerable<string> pathElements = backslashedDosPath.Split(DirectorySeparatorChar, StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
+
+        bool moveNext = false;
+        bool appendedFolder = false;
+        bool mustPrependDirectorySeparator = false;
+        foreach (string pathElement in pathElements) {
+            if(pathElement == ".." && appendedFolder) {
+                moveNext = true;
+            }
+            else {
+                if(moveNext) {
+                    moveNext = false;
+                    continue;
+                }
+                if(pathElement != "." && pathElement != ".." && !pathElement.Contains(VolumeSeparatorChar)) {
+                    appendedFolder = true;
+                    if(mustPrependDirectorySeparator) {
+                        normalizedDosPath.Append(DirectorySeparatorChar);
+                    }
+                    normalizedDosPath.Append(pathElement.ToUpperInvariant());
+                    mustPrependDirectorySeparator = true;
+                }
+            }
+        }
+
+        return ConvertUtils.ToBackSlashPath(normalizedDosPath.ToString());
+    }
+
+    private DosFileOperationResult SetCurrentDirValue(char driveLetter, string? hostFullPath) {
+        if (string.IsNullOrWhiteSpace(hostFullPath) ||
+            !IsWithinMountPoint(hostFullPath, _driveMap[driveLetter]) ||
+            Encoding.ASCII.GetByteCount(hostFullPath) > MaxPathLength) {
+            return DosFileOperationResult.Error(ErrorCode.PathNotFound);
+        }
+
+        _driveMap[driveLetter].FullHostCurrentDirectory = hostFullPath;
+        return DosFileOperationResult.NoValue();
+    }
+
+    /// <summary>
+    /// Converts the DOS path to a full host path of the parent directory.<br/>
+    /// </summary>
+    /// <param name="dosPath">The DOS path to convert.</param>
+    /// <returns>A string containing the full path to the parent directory in the host file system, or <c>null</c> if nothing was found.</returns>
+    public string? TryGetFullHostParentPathFromDos(string dosPath) {
```

Comment:
In general, `TryGet..` methods return a `bool` to indicate success and return the value via an `out` parameter.
The pattern where a default is returned is usually called `Get...OrDefault`

that's just naming nitpicking though

Reply:
Method renamed.